doc/getting_started.rst

```diff
@@ -0,0 +1,185 @@
+
+  >>> from sklearn.preprocessing import StandardScaler
+  >>> from sklearn.linear_model import LogisticRegression
+  >>> from sklearn.pipeline import make_pipeline
+  >>> from sklearn.datasets import load_iris
+  >>> from sklearn.model_selection import train_test_split
+  >>> from sklearn.metrics import accuracy_score
+  ...
+  >>> preprocessor = StandardScaler()
+  >>> predictor = LogisticRegression(random_state=0)
+  >>> pipe = make_pipeline(preprocessor, predictor)
+  ...
+  >>> X, y = load_iris(return_X_y=True)
+  >>> X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+  >>> pipe.fit(X_train, y_train)  # fit the preprocessor first, then the predictor
+  Pipeline(steps=[('standardscaler', StandardScaler()),
+                  ('logisticregression', LogisticRegression(random_state=0))])
+
+  >>> accuracy_score(pipe.predict(X_test), y_test)
+  0.97...
+
+Model evaluation
+----------------
+
+Once a model has been fitted, it is natural to evaluate its performance on
+unseen data. We have just seen the
+:func:`~sklearn.model_selection.train_test_split` helper, but scikit-learn
+provides many other tools for model evaluation, in particular for
+:ref:`cross-validation <cross_validation>`.
+
+We here briefly show how to perform a 3-folds cross-validation using the
+:func:`~sklearn.model_selection.cross_validate` helper. Note that it is also
+possible to manually iterate over the folds, and to use different data
+splitting strategies. Please refer to our :ref:`User Guide
+<cross_validation>` for more details.
+
+  >>> from sklearn.datasets import make_regression
+  >>> from sklearn.linear_model import LinearRegression
+  >>> from sklearn.model_selection import KFold
+  >>> from sklearn.model_selection import cross_validate
+  ...
+  >>> X, y = make_regression(n_samples=1000, random_state=0)
+  >>> lr = LinearRegression()
+  >>> cv = KFold(n_splits=3)
+  ...
+  >>> result = cross_validate(lr, X, y, cv=cv)
+  >>> result['test_score']
+  array([1., 1., 1.])
+
+Automatic parameter searches
+----------------------------
+
+All estimators have parameters that can be tuned for the predictions to be as
+good as possible. For example a
+:class:`~sklearn.ensemble.RandomForestRegressor` has a ``n_estimators``
+parameters that determines the number of trees in the forest, and a
+``max_depth`` parameter that determines the maximum depth of each tree.
+Quite often, it is not clear what the exact values of these parameters should
+be since they depend on the data at hand.
+
+Scikit-learn provides tools to automatically search the best parameter
+combinations (via cross-validation). In the following example, we randomly
+search over the parameter state of a random forest with a
+:class:`~sklearn.model_selection.RandomizedSearchCV` object. When the search
+is over, the :class:`~sklearn.model_selection.RandomizedSearchCV` behaves as
+a :class:`~sklearn.ensemble.RandomForestRegressor` that has been fitted with
```

Comment:
we should use a pipeline instead of a single model, to illustrate how to treat the whole pipeline as an estimator, as well as how to pass grid search parameters of a pipeline.

Reply:
Ugh, I agree but I'm a bit concerned that the `estimator__param` logic is a bit too much/scary for such an introduction.

Would you be OK to add a note that basically says in practice you always want to use a pipeline, let alone for the fact that it prevents data leaks?

Reply:
I don't think it's scary. We may have to add a paragraph or two to explain how it works, but ideally at the end of this tutorial, the user has a code snippet they can copy/paste and use. I know we want to discourage people from copy pasting random code, but that's what they do. Also, if it's too scary for people to pass the pipeline parameters to a gridsearch, then we need to change the API, since that's not gonna happen (any time soon at least) I think we really should have it in the first tutorial.

We also have a specific example for the combination of grid search and pipeline, we can have the code snippet here, with a bit of explanation, and then link to that example.

Reply:
I'll try to come up with something, but I really don't think we should consider the getting-started guide as a tutorial.

The way I see it, the purpose of this guide is to showcase the main features of scikit-learn, ideally with as few cognitive overload as possible. IMO grid searching a pipeline adds a significant cognitive overload when the point is simply to explain grid search.

Reply:
I made this

```py
>>> from sklearn.datasets.california_housing import fetch_california_housing
>>> from sklearn.ensemble import RandomForestRegressor
>>> from sklearn.model_selection import RandomizedSearchCV
>>> from sklearn.pipeline import make_pipeline
>>> from sklearn.preprocessing import StandardScaler
>>> from sklearn.model_selection import train_test_split
>>> from scipy.stats import randint
...
>>> X, y = fetch_california_housing(return_X_y=True)
>>> X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
...
>>> # create a pipeline
>>> pipe = make_pipeline(StandardScaler(), RandomForestRegressor(random_state=0))
...
>>> # define the parameter space that will be searched over. We here only
>>> # search over the parameters of the random forest.
>>> param_distributions = {'randomforestregressor__n_estimators': randint(1, 5),
...                        'randomforestregressor__max_depth': randint(5, 10)}
...
>>> # now create a searchCV object and fit it to the data
>>> search = RandomizedSearchCV(estimator=pipe,
...                             n_iter=5,
...                             param_distributions=param_distributions,
...                             random_state=0)
>>> search.fit(X_train, y_train)
RandomizedSearchCV(estimator=Pipeline(steps=[('standardscaler',
                                              StandardScaler()),
                                             ('randomforestregressor',
                                              RandomForestRegressor(random_state=0))]),
                   n_iter=5,
                   param_distributions={'randomforestregressor__max_depth': ...,
                                        'randomforestregressor__n_estimators': ...},
                   random_state=0)
>>> search.best_params_
{'randomforestregressor__max_depth': 9, 'randomforestregressor__n_estimators': 4}

>>> # the search object now acts like a normal pipeline / estimator
>>> # with max_depth=9 and n_estimators=4
>>> search.score(X_test, y_test)
0.73...
```

I don't really like it because I think it's way too much code for illustrating the grid searching (the current example is already too big IMO), and it's also a bad example because you don't care about scaling when using a RF.

I've tried to think of real-case scenarios e.g. using an imputer, but that requires much more code, and that is irrelevant w.r.t the point of the example which is to illustrate grid search (in its simplest form).

I'm still open to the idea, but unless we can have a simple example, I would be -1. We need simple examples in this guide else users will be discouraged right from the start.

Reply:
related: I added a note in 
a0359a2